vip-helpers/vip-utils.php

```diff
@@ -796,6 +799,30 @@ function vip_safe_wp_remote_get( $url, $fallback_value='', $threshold=3, $timeou
 	return $response;
 }
 
+/**
+ * This is a sophisticated extended version of wp_remote_get(). It is designed to more gracefully handle failure than wpcom_vip_file_get_contents() does.
+ *
+ * Note that like wp_remote_get(), this function does not cache.
+ *
+ * @author tottdev
+ * @link http://vip.wordpress.com/documentation/fetching-remote-data/ Fetching Remote Data
+ * @param string $url URL to fetch
+ * @param string $fallback_value Optional. Set a fallback value to be returned if the external request fails.
+ * @param int $threshold Optional. The number of fails required before subsequent requests automatically return the fallback value. Defaults to 3, with a maximum of 10.
+ * @param int $timeout Optional. Number of seconds before the request times out. Valid values 1-3; defaults to 1.
+ * @param int $retry Optional. Number of seconds before resetting the fail counter and the number of seconds to delay making new requests after the fail threshold is reached. Defaults to 20, with a minimum of 10.
+ * @param array Optional. Set other arguments to be passed to wp_remote_get().
+ * @return string|WP_Error|array Array of results. If fail counter is met, returns the $fallback_value, otherwise return WP_Error.
```

Comment:
Minor: Should we just skip the duplicate `@param` docs here since they're the same as the new function?

Reply:
The docblock is used to generate documentation, so I think if we omitted it here, the docs would be incomplete? Could leave a note to see `vip_safe_wp_remote_request()`, but that feels clunky as a UX for someone reading the docs. Thoughts?

Reply:
It's a pretty common practice to crosslink functions in docs. 

The main concern with duplicating them is that these could get out of date which would be much worse than bad UX.

Reply:
True - switched it out in 70d79679